### PR TITLE
change logic to update user role and fix issue reported in git

### PR DESCRIPTION
**Pull request by CharlotteSpinks**

### CODE REVIEW

File: lib.php [hunk 1]
```diff
@@ -3045,15 +3045,22 @@ function plagiarism_turnitin_send_queued_submissions() {
...
+        $submission->setRole('Learner');
+
+        if ($queueditem->userid != $queueditem->submitter) {
+
             $instructor = new turnitin_user($queueditem->submitter, 'Instructor');
-            $instructor->edit_tii_user();
 
-            $submission->setSubmitterUserId($instructor->tiiuserid);
-            $submission->setRole('Instructor');
+            // This should be true but in case of an edge case where a user has been deleted in Tii.
```
Comment (jmcgettrick) [inline lib.php:3055]: "These"

File: classes/turnitin_user.class.php [hunk 2]
```diff
@@ -299,10 +300,12 @@ public function edit_tii_user() {
             try {
                 $turnitincall->updateUser($user);
             } catch (Exception $e) {
-                $toscreen = ($this->workflowcontext == "cron") ? false : true;
+                $toscreen = ($this->workflowcontext == "cron") == false;
```
[the remark this answers is not in the document]
Comment (CharlotteSpinks) [reply classes/turnitin_user.class.php]: Done.

File: lib.php (hunk 1 above)
Comment (CharlotteSpinks) [reply lib.php]: Done.

File: lib.php [hunk 3]
```diff
@@ -3045,15 +3045,22 @@ function plagiarism_turnitin_send_queued_submissions() {
...
+
+        if ($queueditem->userid != $queueditem->submitter) {
+
             $instructor = new turnitin_user($queueditem->submitter, 'Instructor');
-            $instructor->edit_tii_user();
 
-            $submission->setSubmitterUserId($instructor->tiiuserid);
-            $submission->setRole('Instructor');
+            // These should be true but in case of an edge case where a user has been deleted in Tii.
+            $updatesuccess = $instructor->edit_tii_user();
```
Comment (dwinn) [inline lib.php:3056]: These two variables are only used once so I'm not sure we need them. Could just do:

`if ($instructor->edit_tii_user() && $instructor->join_user_to_class($coursedata->turnitin_cid))`

Or if the variables must stay then let's give them better names, such as $isupdated and $isenrolled. It's good practise to prefix boolean variables with `is` or `has`. See https://dev.to/michi/tips-on-naming-boolean-variables-cleaner-code-35ig

Comment (CharlotteSpinks) [reply lib.php]: Yeah this makes sense, I've gone with the first option. Thanks for the variable naming link :D

**Pull request merged**